kokoro/linux/build.sh

```diff
@@ -81,6 +81,46 @@ do
 done
 echo $(date): Smoketests completed.
 
+set -e
+
 # Build the release packages.
 mkdir $BUILD_ROOT/out
 $SRC/kokoro/linux/package.sh $BUILD_ROOT/out
+
+##
+## Test capture and replay of the Vulkan Sample App.
+##
+
+# Install the Vulkan loader and xvfb (X virtual framebuffer).
+sudo apt-get -qy install libvulkan1 xvfb
+
+# Get prebuilt SwiftShader.
+curl -fsSL -o swiftshader.zip https://github.com/google/gfbuild-swiftshader/releases/download/github%2Fgoogle%2Fgfbuild-swiftshader%2F0bbf7ba9f909092f0328b1d519d5f7db1773be57/gfbuild-swiftshader-0bbf7ba9f909092f0328b1d519d5f7db1773be57-Linux_x64_Debug.zip
+unzip -d swiftshader swiftshader.zip
+
+# Use SwiftShader.
+export VK_ICD_FILENAMES="$(pwd)/swiftshader/lib/vk_swiftshader_icd.json"
+export VK_LOADER_DEBUG=all
+
+# Just try running the app first.
+
+# Allow non-zero exit status.
+set +e
+
+xvfb-run -e xvfb.log -a timeout --preserve-status -s INT -k 10 5 bazel-bin/cmd/vulkan_sample/vulkan_sample
+
+APP_EXIT_STATUS="${?}"
+
+set -e
+
+if test -f xvfb.log; then
+  cat xvfb.log
+fi
+
+# This line will exit with status 1 if the app's exit status
+# was anything other than 130 (128+SIGINT).
+test "${APP_EXIT_STATUS}" -eq 130
+
+xvfb-run -e xvfb.log -a bazel-bin/pkg/gapit trace -device host -disable-coherentmemorytracker -disable-pcs -disable-unknown-extensions -record-errors -no-buffer -api vulkan -start-at-frame 5 -capture-frames 10 -observe-frames 1 -out vulkan_sample.gfxtrace bazel-bin/cmd/vulkan_sample/vulkan_sample
+
+xvfb-run -e xvfb.log -a bazel-bin/pkg/gapit video -gapir-nofallback -type sxs -frames-minimum 3 -out vulkan_sample.mp4 vulkan_sample.gfxtrace
```

Comment:
I dont think you need -type sxs

Reply:
Also why do we have -frames-minumum? Don't expect exactly 10 frames ?

Reply:
Do you mean it should be `-frames-minimum 10`? Sounds fair to me. I think 3 is just to safeguard against the case where no frames were captured. But you would hope `trace` would fail in this case. And indeed, it would be bad if only 3 frames were captured.

Without "-type sxs", I think the default will usually be sxs anyway, but it may switch to regular video if no framebuffer observations were found. Using "-type sxs" ensures we fail if there are no framebuffer observations (although we have a safeguard against this already!).

Reply:
I changed to `-frames-minimum 10`